build_tools/ci/run_matmul_test.sh

```diff
@@ -120,12 +144,28 @@ function run_matmul_test() {
       --iree-amd-aie-peano-install-dir=${peano_install_path} \
       --iree-amd-aie-mlir-aie-install-dir=${mlir_aie_install_path} \
       --iree-amd-aie-vitis-install-dir=${vitis_path} \
+      --iree-hal-dump-executable-files-to=$PWD \
+      --iree-hal-dump-executable-intermediates-to=$PWD \
       -o "${OUTPUT_DIR}/${name}_matmuls.vmfb"
   ${IREE_COMPILE_EXE} \
       "${OUTPUT_DIR}/${name}_calls.mlir" \
       --iree-hal-target-backends=${target_backend} \
       -o "${OUTPUT_DIR}/${name}_calls.vmfb"
 
+  # Extract function names from the mlir file
+  function_names=$(grep -oP '@\K\S+(?=\()' ${OUTPUT_DIR}/${name}_matmuls.mlir)
+
+  # Iterate over each function name and sign the corresponding XCLBIN
```

Comment:
Not sure what this is doing / why this is needed... but this looks like something that the tools should do on their own, rather than need some extra scripting. Fine for now but wouldn't want this to spread to many other files.

The `iree-hal-dump-*` flags are for developer use and should not be load bearing for a deployment workflow.

Reply:
Ya this is a signing process needed for running on AIE hardware at the moment, the estimate it is that we wont be needing it in about a month.

Reply:
If it were merged into the run-module tool, either that must be run as root or it has to perform operations in a directory that is normally configured with root permissions.
Nobody likes this part, you don't need root powers to run things on GPU, but this is where we're at...
I proposed creating a separate tool for this purpose, but it's not a priority since apparently this requirement is expected to go away.